crates/fuel-core/src/service/sub_services/algorithm_updater.rs

```diff
@@ -193,8 +188,6 @@ pub fn get_synced_gas_price_updater(
     settings: ConsensusParametersProvider,
```

Comment:
I think that most of the logic of this function should live inside the service itself. Only a small part of the `sync_metadata_storage_with_on_chain_storage` logic related to the on-chain database should live here

Reply:
I agree ~ this is part of the follow up PR though, which sunsets `InitializeTask` appropriately.